File: Completed/Go/111.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func minDepth(root *TreeNode) int {
    depth := 1<< 63-1
    
    helper(root, 1, &depth)
    
    if depth == 1<< 63-1 {
        return 0
    }
    return depth
}

func helper(root *TreeNode, depth int, minDepth *int) {
    if root == nil {
        return
    }
    
    if root.Left == nil && root.Right == nil && depth < *minDepth {
        *minDepth = depth
        return
    }
    
    helper(root.Left, depth+1, minDepth)
    helper(root.Right, depth+1, minDepth)
}

File: Completed/python3/111.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: TreeNode) -> int:
      if root == None:
        return 0
      return self.helper(root)
    def helper(self, root:TreeNode) -> int:
      if root.left == None and root.right == None:
        return 1
      
      val = 1<<63
      if root.left != None:
        val = min(val, self.helper(root.left)+1)
      if root.right != None:
        val = min(val, self.helper(root.right)+1)
        
      return val
